pkg/services/policer/process.go

```diff
@@ -55,8 +55,8 @@ func (p *Policer) shardPolicyWorker(ctx context.Context) {
 						return
 					}
 
-					p.processObject(ctx, addr)
 					p.cache.Add(addrStr, time.Now())
+					p.processObject(ctx, addr)
```

Comment:
I think it is worth to also put `time.Now()` after this line. Timeout should start after we have finished processing.

Reply:
Yes.

Reply:
updated a little: added the map that contains objects that are already being processed (max map size is a max pool size) while LRU cache is gonna store object that has been processed recently (200_000 MiB size). originally suggested by @cthulhu-rider 

what do you think?

Reply:
To me 2 maps increase memory consumption, can we reuse the same map for this?
Like store `0` if the processing is in progress, before submitting the task and update with the actual value afterwise.

Reply:
isn't it useless to prevent concurrent object handling via LRU cache? i mean if it is critical for us not to handle objects twice at a time

in fact, i am a little confused about that issue: we are trying to solve problems that we are not going to face in real use-cases (using a node for storing the only one object)